### ISSUE TITLE
Add Java 21

### ISSUE THREAD

**Issue by Madjosz** [codewars/runner#273]
- Name: Java
- Version: [21 (LTS)](https://openjdk.org/projects/jdk/21/)
- Release Note/Changelog:
  - [Simple Web Server](https://openjdk.org/jeps/408)
  - [Pattern Matching for `switch`](https://openjdk.org/jeps/441)  and [records](https://openjdk.org/jeps/440)
  - [Virtual Threads](https://openjdk.org/jeps/444)
  - [Sequenced Collections](https://openjdk.org/jeps/431)
---

:+1: reaction _might_ help to get this request prioritized.

**Comment by Feliks-WR** [comment]
PLEASE DO!

**Comment by MoonBox5** [comment]
IMHO Java is so popular in the real world that its users on Codewars expect LTS updates without much fuss. To me it's a little strange that this must even be said, but we've already gone more than a year with no Java 21.
Kindly update to 21 LTS please. A lot of us are excited for pattern matching features for example.

**Comment by tweimer** [comment]
I am waiting for it

**Comment by AndrzejProchyra** [comment]
Perhaps it makes more sense to update to [Java 25](https://openjdk.org/projects/jdk/25/) now. That is the current LTS release since September 2025. Java 17 is over 4 years old now.

**Comment by Feliks-WR** [comment]
Yeah, time for Java 25.

**Comment by Feliks-WR** [comment]
> Perhaps it makes more sense to update to [Java 25](https://openjdk.org/projects/jdk/25/) now. That is the current LTS release since September 2025. Java 17 is over 4 years old now.

I doubt they'll do anything, I just left the platform

**Comment by Kacarott** [comment]
> I just left the platform

I'd encourage you to send an email to info@codewars.com, telling them exactly why you are leaving. Maybe if they realise that people are leaving it will get them to actually do something... ☹️

**Comment by Feliks-WR** [comment]
> > I just left the platform
> 
> I'd encourage you to send an email to [info@codewars.com](mailto:info@codewars.com), telling them exactly why you are leaving. Maybe if they realise that people are leaving it will get them to actually do something... ☹️

Yeah, I just don't care about the platform anymore. For a long time, they have used old versions, it's just not fun.

**Comment by hobovsky** [comment]
Java has been updated to 21.

**Comment by tweimer** [comment]
I do not see it available for any Kata I checked.

**Comment by BobtheLantern** [comment]
Very few java kata have been updated to 21 so far. The language version is available, though, so you can always help by updating the kata yourself!
